### PR TITLE
Support passing subscriptionId and resourceGroupName to createIoTHub API

### PR DESCRIPTION
**Pull request by formulahendry**

### CODE REVIEW

File: src/iotHubResourceExplorer.ts [hunk 1]
```diff
@@ -257,6 +256,26 @@ export class IoTHubResourceExplorer extends BaseExplorer {
         });
     }
 
+    private async getOrSelectSubscriptionItem(outputChannel: vscode.OutputChannel, subscriptionId: string): Promise<SubscriptionItem> {
+        if (subscriptionId) {
+            let azureResourceFilter = this.accountApi.filters.find((filter) => filter.subscription.subscriptionId === subscriptionId);
```
Comment (adashen) [inline src/iotHubResourceExplorer.ts]: is the filter for active subs or all subs?

Comment (formulahendry) [reply src/iotHubResourceExplorer.ts]: The filter is for active subs. Change to all subs now.

File: src/iotHubResourceExplorer.ts [hunk 2]
```diff
@@ -130,27 +130,26 @@ export class IoTHubResourceExplorer extends BaseExplorer {
...
             return;
         }
         TelemetryClient.sendEvent("General.Select.Subscription.Start");
-        const subscriptionItems = this.loadSubscriptionItems(this.accountApi);
-        const subscriptionItem = await vscode.window.showQuickPick(subscriptionItems, { placeHolder: "Select Subscription", ignoreFocusOut: true });
-        if (subscriptionItem) {
-            TelemetryClient.sendEvent("General.Select.Subscription.Done");
+        const subscriptionItem = await this.getOrSelectSubscriptionItem(outputChannel, subscriptionId);
+        if (!subscriptionItem) {
+            return;
```
Comment (adashen) [inline src/iotHubResourceExplorer.ts]: should return or throw error?

File: src/azure-account.api.d.ts [hunk 3]
```diff
@@ -1,10 +1,13 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT license.
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
```
[the remark this answers is not in the document]
Comment (formulahendry) [reply src/azure-account.api.d.ts]: Just copy from https://github.com/Microsoft/vscode-azure-account/blob/master/src/azure-account.api.d.ts

File: src/iotHubResourceExplorer.ts (hunk 2 above)
Comment (formulahendry) [reply src/iotHubResourceExplorer.ts]: `return` indicates user cancelled or subscription not found.

**Pull request merged**